internal/core/metadata/rest_deviceprofile.go

```diff
@@ -360,7 +366,7 @@ func deleteDeviceProfile(dp models.DeviceProfile, w http.ResponseWriter) error {
 		return err
 	}
 	if len(pw) > 0 {
-		err := errors.New("Cant delete device profile, the profile is still in use by a provision watcher")
+		err = errors.New("Cant delete device profile, the profile is still in use by a provision watcher")
```

Comment:
Is it worth creating custom error types and using those here? See internal/core/metadata/errors/types.go

We'll probably do that anyway at some point when we decouple biz logic from these router files.